### PR TITLE
Add Google Spreadsheet Example

### PR DESCRIPTION
**Pull request by isuruuy429**
This PR contains adding the Google Spreadsheet Connector example. 
(The images for the /create /read and /edit resources are yet to be added)

Resolves: https://github.com/wso2/micro-integrator/issues/1557

### CODE REVIEW

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+
+It has been provided the ability to do Spreadsheet operations and Spreadsheet Data operations. 
+
+## What you'll build
+
+This example explains how to use Google Spreadsheet Connector to create a Google spreadsheet, write data to it and get them read. Further, it explains how the data in the spreadsheet can be edited. 
+
+It will have three HTTP API resources, which are `insert`, `read` and `edit`. 
+
+* `/insert `: The user sends the request payload which includes the name of the spreadsheet should be, what are the sheet names and what data should be inserted to which sheet and the which range of cells.  This request is sent to WSO2 EI by invoking the Spreadsheet API. It creates a spreadsheet with specified data in the specified cell rage.
```
Comment (abeykoon) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:13]: "to which sheet and the which range of cells" some grammar issue.

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Get Credentials for Google Spreadsheet
+
+To obtain the access token, client Id, client secret and refresh token, we need to follow the below steps. 
+
+1. Open the [Google API Console Credentials](https://console.developers.google.com/apis/credentials) page. You will be prompt to log in to a Google Account. Log in to your relevant Google Account.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:5]:
```suggestion
1. Open the [Google API Console Credentials](https://console.developers.google.com/apis/credentials) page. You will be prompted to log in to a Google Account. Log in to your relevant Google Account. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  <img src="../../../../assets/img/connectors/credentials.png" title="Credentials" width="800" alt="Credentials" />
+
+9. Click Library on the side menu, search for **Google Sheets API** and click on it.
+
+10. Click **Enable** to enable the Google Sheets API.
+  <img src="../../../../assets/img/connectors/sheetsapi.png" title="Enable Google Sheets API" width="800" alt="Enable Google Sheets API" />
+
+
+## Obtaining Access Token and Refresh Token
+1. Navigate to [OAuth 2.0 Playground](https://developers.google.com/oauthplayground/) and click OAuth 2.0 Configuration button in the Right top corner.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:40]:
```suggestion
1. Navigate to [OAuth 2.0 Playground](https://developers.google.com/oauthplayground/) and click the OAuth 2.0 Configuration button in the top right corner of your screen.
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+2. Select **Use your own OAuth credentials**, and provide the obtained Client ID and Client Secret values as above click on Close.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:42]:
```suggestion
2. Select **Use your own OAuth credentials**, and provide the obtained Client ID and Client Secret values. Click on **Close**.
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  <img src="../../../../assets/img/connectors/oath-configuration.png" title="Obtaining Oauth-configuration" width="800" alt="Obtaining Oauth-configuration" />
+
+3. Under Step 1, select `Google Sheets API v4` from the list of APIs, select all the scopes.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:45]:
```suggestion
3. Under Step 1, select `Google Sheets API v4` from the list of APIs and select all the scopes. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+2. Select **Use your own OAuth credentials**, and provide the obtained Client ID and Client Secret values as above click on Close.
+  <img src="../../../../assets/img/connectors/oath-configuration.png" title="Obtaining Oauth-configuration" width="800" alt="Obtaining Oauth-configuration" />
+
+3. Under Step 1, select `Google Sheets API v4` from the list of APIs, select all the scopes. 
+  <img src="../../../../assets/img/connectors/sheetsapi2.png" title="Selecting Scopes" width="800" alt="Selecting Scopes" />
+
+4. Click on **Authorize APIs** button and select your Gmail account when you are asked and allow the scopes.
+  <img src="../../../../assets/img/connectors/sheetsapi4.png" title="Grant Permission" width="800" alt="Grant Permission" />
+
+5.  Under Step 2, click **Exchange authorization code for tokens** to generate an display the Access Token and Refresh Token. Now we are done with configuring the Google Sheets API.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:51]:
```suggestion
5.  Under Step 2, click **Exchange authorization code for tokens** to generate and display the Access Token and Refresh Token. Now we are done with configuring the Google Sheets API.
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
+# Get Credentials for Google Spreadsheet
+
+To obtain the access token, client Id, client secret and refresh token, we need to follow the below steps.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:3]:
```suggestion
To obtain the Access Token, Client Id, Client Secret and Refresh Token, we need to follow the below steps. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+        "spreadsheetId": "1Ht0FWeKtKqBb1pEEzLcRMM8s5mktJdhivX3iaFXo-qQ",
+        "updatedRange": "Employees!A6:C7",
+        "updatedRows": 2,
+        "updatedColumns": 3,
+        "updatedCells": 6
+      }
+    ]
+  }
+```
+The spreadsheet should be edited with in the above specified cell range.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:382]:
```suggestion
The spreadsheet should be edited within the above specified cell range. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 9]
```diff
@@ -0,0 +1,387 @@
...
+        "Female"
+      ]
+    ]
+  }
+
+```
+
+### Spreadsheet Edit Operation
+
+1. Invoke the SpreadsheetAPI with the following URL. Application such as [Postman](https://www.postman.com/) can be used to invoke the API. Obtain the Spreadsheet ID from the step 1.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:343]:
```suggestion
1. Invoke the SpreadsheetAPI with the following URL. An application such as [Postman](https://www.postman.com/) can be used to invoke the API. Obtain the Spreadsheet ID from step 1. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 10]
```diff
@@ -0,0 +1,387 @@
...
+      "updatedRows": 3,
+      "updatedColumns": 3,
+      "updatedCells": 9
+    }
+  }
+```
+
+### Spreadsheet Read Operation
+
+1. Invoke the SpreadsheetAPI with the following URL. Application such as [Postman](https://www.postman.com/) can be used to invoke the API. Obtain the Spreadsheet ID from the step 1.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:301]:
```suggestion
Invoke the SpreadsheetAPI with the following URL. Application such as [Postman](https://www.postman.com/) can be used to invoke the API. Obtain the Spreadsheet ID from the step 1. 
```
Remove the list numbering if there is only one item. Please apply below as well.

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+
+Follow these steps to deploy the exported CApp in the Enterprise Integrator Runtime. 
+
+{!references/connectors/deploy-capp.md!}
+
+## Testing
+
+### Spreadsheet insert Operation
+
+1. Invoke the SpreadsheetAPI with the following URL. Application such as [Postman](https://www.postman.com/) can be used to invoke the API.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:236]:
```suggestion
Invoke the SpreadsheetAPI with the following URL. An application such as [Postman](https://www.postman.com/) can be used to invoke the API. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+
+2. Right click on the created Integration Project and select, -> **New** -> **Rest API** to create the REST API. 
+    <img src="../../../../assets/img/connectors/adding-an-api.png" title="Adding a Rest API" width="800" alt="Adding a Rest API"/>
+
+3. Provide the API name as `SpreadsheetAPI` and the API context as `/insert`.
+
+4. First we will create the `/insert` resource. Right click on the API Resource and go to **Properties** view. We use a URL template called `/insert` as we have two API resources inside single API. The method will be `Post`. 
+    <img src="../../../../assets/img/connectors/filecon-3.png" title="Adding the API resource." width="800" alt="Adding the API resource."/>
+
+5. In this operation we are going to receive input from the user which are `properties`, `sheets`, `range` and `values`.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:40]:
```suggestion
5. In this operation we are going to receive input from the user, which are `properties`, `sheets`, `range` and `values`. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 13]
```diff
@@ -0,0 +1,387 @@
...
+4. First we will create the `/insert` resource. Right click on the API Resource and go to **Properties** view. We use a URL template called `/insert` as we have two API resources inside single API. The method will be `Post`. 
+    <img src="../../../../assets/img/connectors/filecon-3.png" title="Adding the API resource." width="800" alt="Adding the API resource."/>
+
+5. In this operation we are going to receive input from the user which are `properties`, `sheets`, `range` and `values`. 
+    - properties - It can provide the spreadsheet properties such as title of the spreadsheet. 
+    - sheets - It can provide set of sheets to be created.
+    - range - It provides the sheet name and the range that data need to be inserted. 
+    - values - Data to be inserted.
+
+6. The above four parameters are saved to a property group. Drag and drop the Property Group mediator onto the canvas in the design view and do as shown below. For further reference, you can read about the [Property Group mediator](https://ei.docs.wso2.com/en/latest/micro-integrator/references/mediators/property-Group-Mediator/). You can add set of properties as below.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:46]: Let's make this a relative link.

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 14]
```diff
@@ -0,0 +1,387 @@
...
+            <refreshToken></refreshToken>
+        </googlespreadsheet.init>
+        <googlespreadsheet.createSpreadsheet>
+            <properties>{$ctx:properties}</properties>
+            <sheets>{$ctx:sheets}</sheets>
+        </googlespreadsheet.createSpreadsheet>
+    </sequence>
+```
+
+12. Next we need to create the `addData.xml` sequence as above. As explained in step 8, create a sequence by right clicking the Integration Project which has already been created.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:87]:
```suggestion
12. Next we need to create the `addData.xml` sequence as above. As explained in step 8, create a sequence by right clicking the Integration Project that has already been created. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 15]
```diff
@@ -0,0 +1,387 @@
...
+* `/read `: The user sends the request payload, which includes the spreadsheet Id which should be obtained from calling the `insert` API resource, and the range of the cell range to be read. 
+
+* `/edit `: The user sends the request payload which includes the spreadsheet Id which should be obtained from calling the `insert` API resource, and the data to be edited that includes values and the range. 
+
+
+If you do not want to configure this yourself, you can simply [get the project](#get-the-project) and run it.
+
+## Configure the connector in WSO2 Integration Studio
+
+Follow these steps to set up the ESB Solution Project and the Connector Exporter Project.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:24]:
```suggestion
Follow these steps to set up the Integration Project and the Connector Exporter Project. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 16]
```diff
@@ -0,0 +1,387 @@
...
+
+This example explains how to use Google Spreadsheet Connector to create a Google spreadsheet, write data to it and get them read. Further, it explains how the data in the spreadsheet can be edited. 
+
+It will have three HTTP API resources, which are `insert`, `read` and `edit`. 
+
+* `/insert `: The user sends the request payload which includes the name of the spreadsheet should be, what are the sheet names and what data should be inserted to which sheet and the which range of cells.  This request is sent to WSO2 EI by invoking the Spreadsheet API. It creates a spreadsheet with specified data in the specified cell rage. 
+
+* `/read `: The user sends the request payload, which includes the spreadsheet Id which should be obtained from calling the `insert` API resource, and the range of the cell range to be read. 
+
+* `/edit `: The user sends the request payload which includes the spreadsheet Id which should be obtained from calling the `insert` API resource, and the data to be edited that includes values and the range.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:17]:
```suggestion
* `/edit `: The user sends the request payload, which includes the spreadsheet Id that should be obtained from calling the `insert` API resource, and the data to be edited that includes values and the range. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md [hunk 17]
```diff
@@ -0,0 +1,387 @@
...
+
+## What you'll build
+
+This example explains how to use Google Spreadsheet Connector to create a Google spreadsheet, write data to it and get them read. Further, it explains how the data in the spreadsheet can be edited. 
+
+It will have three HTTP API resources, which are `insert`, `read` and `edit`. 
+
+* `/insert `: The user sends the request payload which includes the name of the spreadsheet should be, what are the sheet names and what data should be inserted to which sheet and the which range of cells.  This request is sent to WSO2 EI by invoking the Spreadsheet API. It creates a spreadsheet with specified data in the specified cell rage. 
+
+* `/read `: The user sends the request payload, which includes the spreadsheet Id which should be obtained from calling the `insert` API resource, and the range of the cell range to be read.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:15]:
```suggestion
* `/read `: The user sends the request payload, which includes the spreadsheet Id that should be obtained from calling the `insert` API resource, and the range of the cell range to be read. 
```

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 1 above)
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:13]:
```suggestion
* `/insert `: The user sends the request payload, which includes the name of the spreadsheet, the sheet names, and what data should be inserted to which sheet and which range of cells.  This request is sent to WSO2 EI by invoking the Spreadsheet API. It creates a spreadsheet with specified data in the specified cell range. 
```

Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:13]: Changed

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md (hunk 2 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:5]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md (hunk 3 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:40]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md (hunk 4 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:42]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md (hunk 5 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:45]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md (hunk 6 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:51]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md (hunk 7 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/get-credentials-for-google-spreadsheet.md:3]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 8 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:382]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 9 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:343]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 10 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:301]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 11 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:236]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 12 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:40]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 14 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:87]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 15 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:24]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 16 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:17]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 17 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:15]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 1 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:13]: Done

File: en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md (hunk 13 above)
Comment (isuruuy429) [reply en/micro-integrator/docs/references/connectors/google-spreadsheet-connector/google-spreadsheet-connector-example.md:46]: Done

**Pull request merged**